shared-bindings/bleio/Adapter.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Artur Pacholec
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/objproperty.h"
+#include "shared-bindings/bleio/Adapter.h"
+
+//| .. currentmodule:: bleio
+//|
+//| :class:`Adapter` --- BLE adapter information
+//| ----------------------------------------------------
+//|
+//| Get current status of the BLE adapter
+//|
+//| Usage::
+//|
+//|    import bleio
+//|    bleio.adapter.enabled = True
+//|    print(bleio.adapter.address)
+//|
+
+//| .. class:: Adapter()
+//|
+//|     You cannot create an instance of `bleio.Adapter`.
+//|     Use `bleio.adapter` to access the sole instance available.
+//|
+
+//|     .. attribute:: adapter.enabled
+//|
+//|         State of the BLE adapter.
+//|
+
+//|     .. attribute:: adapter.address
+//|
+//|         MAC address of the BLE adapter. (read-only)
+//|
+
+#define BLE_ADDRESS_LEN 17
+
+STATIC mp_obj_t bleio_adapter_get_enabled(mp_obj_t self) {
+    const bool enabled = common_hal_bleio_adapter_get_enabled();
+
+    return enabled ? mp_const_true : mp_const_false;
```

Comment:
You can do `return mp_obj_new_bool(common_hal_bleio_adapter_get_enabled());`. This is common in other `shared-bindings` implementations.

Reply:
Didn't know that one, will fix!